cabal-install/Distribution/Client/Config.hs

```diff
@@ -601,10 +602,11 @@ configFieldDescriptions =
        [simpleField "compiler"
           (fromFlagOrDefault Disp.empty . fmap Text.disp) (optional Text.parse)
           configHcFlavor (\v flags -> flags { configHcFlavor = v })
-        -- TODO: The following is a temporary fix. The "optimization" field is
-        -- OptArg, and viewAsFieldDescr fails on that. Instead of a hand-written
-        -- hackaged parser and printer, we should handle this case properly in
-        -- the library.
+        -- TODO: The following is a temporary fix. The "optimization"
```

Comment:
Indeed. Someone needs to redesign something here and as per usual no one has the time.